sympy/stats/crv.py

```diff
@@ -173,11 +176,26 @@ def check(*args):
 
     def sample(self, size=()):
         """ A random realization from the distribution """
-        icdf = self._inverse_cdf_expression()
         if not size:
-            return icdf(random.uniform(0, 1))
+            size = 1
+        if hasattr(self,'_sample_python'):
+            samp = self._sample_python(size)
+            return samp if size != 1 else samp[0]
+        elif hasattr(self,'_sample_numpy') and import_module('numpy'):
+            samp = self._sample_numpy(size)
+            return samp if size != 1 else samp[0]
+        elif hasattr(self,'_sample_scipy') and import_module('scipy'):
+            samp = self._sample_scipy(size)
+            return samp if size != 1 else samp[0]
+        elif hasattr(self,'_sample_pymc3') and import_module('pymc3'):
+            samp = self._sample_pymc3(size)
+            return samp if size != 1 else samp[0]
```

Comment:
What we if we want to add more libraries in future, like tensorflow? 
Can we generalise this using a loop and storing the library names in a list? I hope you are getting my point.

Reply:
I think you mean that will this be able to extend the `if, elif` ladder by adding such libraries to this ladder and its corresponding `_sample_library` in its class. According to the current API design that if someone needs to add sampling methods to this from any library, add the library to the ladder, then corresponding method to the class and will work.

Reply:
```
        if hasattr(self, '_sample_python'):
            samp = self._sample_python(size)
            return samp if size != 1 else samp[0]

        libraries = ['numpy', 'scipy', 'pymc3']
        for lib in libraries:
            if hasattr(self, '_sample_' + lib ) and import_module(lib):
                samp_method = getattr(self, '_sample_' + lib)
                samp = samp_method(size)
                return samp if size != 1 else samp[0]
```
Do you mean this way? This could also be used. As when in future we add more libraries, we just need to add the library name in list and its corresponding method in the class. Does this sound good?

Reply:
Yeah, this kind of generalisation will be better.

Reply:
Okay, then I will change accordingly